File: Python/cribleErastotene.py
# Fonction Crible d'Eratosthène en Python
# @author Mahier Loïc & PHALAVANDISHVILI Demtre

import time

def trouvePremiers(n):
    ensembles = set()
    nbPremiers = set()
    i = 2
    while i < n :
        if i not in ensembles :
            ensembles.update(range(i * i, n + 1, i))
            nbPremiers.add(i)
        i = i+1
    return nbPremiers

t_avant = time.time()
print(sorted(trouvePremiers(15)))
t_apres = time.time()
print("temps d'execution : ",t_apres - t_avant)

File: Julia/cribleErastotene.jl
# Fonction Crible d'Eratosthène en Julia (non-optimal !!)
# @author Mahier Loïc & PHALAVANDISHVILI Demtre

function afficherPremier(n)
    return [ a for a in 2:n if lesbooleans[a] == true ]
end

function elimineMultiples(x,n)
    for i in (x):(n-x)
        if lesbooleans[i+x]
            if i%x == 0
                lesbooleans[i+x] = false
            end
        end
    end
end

function trouvePremier(n)
    for i in 2:n
        if lesbooleans[i]
            elimineMultiples(i,n)
        end
    end
end

function prog_principale(n)
    global lesbooleans = fill(true,n)
    println("affichage de 1 a ",n)
    println("-debut traitement")
    tdebut = time()
    trouvePremier(n)
    tfin = time()
    println(afficherPremier(n))
    println("-fin traitement - tps traitement : ",tfin-tdebut," s")
end

prog_principale(15)
